server/routes.go

```diff
@@ -212,50 +214,46 @@ func GenerateHandler(c *gin.Context) {
 	checkpointLoaded := time.Now()
 
 	var prompt string
-	var promptVars PromptVars
 	switch {
 	case req.Raw:
 		prompt = req.Prompt
 	case req.Prompt != "":
-		if req.Template != "" {
-			// override the default model template
-			model.Template = req.Template
+		if req.Template == "" {
+			req.Template = model.Template
```

Comment:
Looks like this is setting the req.Template as the "source of truth"? If that is the case you should update the calls to `Prompt()` to use `req.Template` rather than `model.Template`

Reply:
This was definitely problematic, fixed now